cfpb/cfgov-refresh
# Adding Image inset to the full-width molecule

Short description explaining the high-level reason for the pull request

## Additions

- Added `cfgov/jinja2/v1/_includes/molecules/image-inset.html` to render the image inset molecule. Key thing is that we are using the srcset and sizes attribute on the image.  We also use Wagtail to downsize image and create an image for use on mobile. 

- Added  `cfgov/v1/atomic_elements/molecules.py` to render the `Image Inset` molecule within in the `FullTextWidth` organism.  Key thing is the image drop down which will determine the size of the inset.



## Changes

- Modified `cfgov/unprocessed/css/molecules/inset.less` to render the inset images molecule at various breakpoints. Key point is that we never stretch or upscale the image using CSS. We also ensure that the decorative image looks the same as the regular image.

- Modified `cfgov/jinja2/v1/_includes/organisms/full-width-text.html` to ensure the molecule doesn't get wrapped in a block div.

- Modified `cfgov/unprocessed/css/organisms/full-width-text-group.less` to add margins where appropriate at various breakpoints and float modifiers.


## Testing

- Create a `Blog` page in Wagtail.
- Add the `FullTextOrganism` to the `Blog` page.
- Add the `Content` module to the `FullWidthText` organism, and enter some text.
- Add the `Image Inset` organism to the `FullWidthText`.
- Add the `Content` module to the `FullWidthText` organism, and enter some text.
- Experiment with various options within the `Image Inset` interface.
- Preview and verify the image inset works as expected.

## Screenshots

<img width="822" alt="screen shot 2017-01-13 at 3 16 27 am" src="https://cloud.githubusercontent.com/assets/1696212/21922903/1edd0f8a-d93f-11e6-9827-4e5663f27ddc.png">
<img width="397" alt="screen shot 2017-01-25 at 5 10 29 pm" src="https://cloud.githubusercontent.com/assets/1696212/22311365/30c6b6d8-e321-11e6-81d3-9272a4d23dda.png">


## Notes

- The inset macro is a little complicated; the code can be migrated to Django if people feel it's necessary.

- Might consider adding a bottom border option to Wagtail.

- Might consider adding the ability to select a mobile inset image to Wagtail.

## Todos

-

## Checklist
